### PR TITLE
add .mjs (ES6 module) extension to init-js2-mode

### PR DESCRIPTION
**Pull request by mangosmoothie**
adding .mjs extension, which is used by ES6 modules, to the init of the javascript layer.

### CODE REVIEW

File: layers/+lang/javascript/packages.el [hunk 1]
```diff
@@ -102,7 +102,7 @@
     :defer t
     :init
     (progn
-      (add-to-list 'auto-mode-alist '("\\.js\\'" . js2-mode))
+      (add-to-list 'auto-mode-alist '("\\.[m]*js\\'" . js2-mode))
```
Comment (aaronjensen) [inline layers/+lang/javascript/packages.el]: Unless `mmjs` is meant to be valid, this should probably be `m?js`

Comment (mangosmoothie) [reply layers/+lang/javascript/packages.el]: one m is probably sufficient

**Comment by aaronjensen** [comment]
Ready to be merged! (@syl20bnr @TheBB @d12frosted @bmag @JAremko)

**Comment by sdwolfz** [comment]
Thank you :heart:! 
And congratulations on your first Spacemacs PR :tada:!
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**